src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts

```diff
@@ -0,0 +1,367 @@
+  public context: string;
+  // Controls general
+  public id: AbstractControl;
+  public name: AbstractControl;
+  public description: AbstractControl;
+  public entityType: AbstractControl;
+  public entityID: AbstractControl;
+  // Connector
+  public connectorTypeMap = CONNECTOR_TYPE_SELECTION_MAP;
+  public connectorPowerValue: AbstractControl;
+  public connectorPowerUnit: AbstractControl;
+  // Static Restrictions
+  public staticRestrictions!: FormGroup;
+  public connectorType!: AbstractControl;
+  public validFrom: AbstractControl;
+  public validTo: AbstractControl;
+  public minDate: Date;
+  public connectorPowerEnabled!: AbstractControl;
+  // Dimensions
+  public dimensions!: FormGroup;
+  public dimensionsMap: PricingDimensions;
+  public dimensionsKeys: string[];
+  // Flat fee
+  public flatFee: FormGroup;
+  public flatFeeEnabled: AbstractControl;
+  public flatFeeValue: AbstractControl;
+  public flatFeeUnit: AbstractControl;
+  // Energy
+  public energy: FormGroup;
+  public energyEnabled: AbstractControl;
+  public energyValue: AbstractControl;
+  public energyUnit: AbstractControl;
+  public energyStepEnabled: AbstractControl;
+  public energyStepValue: AbstractControl;
+  public energyStepUnit: AbstractControl;
+
+  // Charging time
+  public chargingTime: FormGroup;
+  public chargingTimeEnabled: AbstractControl;
+  public chargingTimeValue: AbstractControl;
+  public chargingTimeUnit: AbstractControl;
+  // Parking time
+  public parkingTime: FormGroup;
+  public parkingTimeEnabled: AbstractControl;
+  public parkingTimeValue: AbstractControl;
+  public parkingTimeUnit: AbstractControl;
+  // Step size
+  public chargingTimeStepEnabled: AbstractControl;
+  public chargingTimeStepValue: AbstractControl;
+  public chargingTimeStepUnit: AbstractControl;
+  public parkingTimeStepEnabled: AbstractControl;
+  public parkingTimeStepValue: AbstractControl;
+  public parkingTimeStepUnit: AbstractControl;
+
+  // eslint-disable-next-line no-useless-constructor
+  public constructor(
+    private centralServerService: CentralServerService,
+    private messageService: MessageService,
+    private spinnerService: SpinnerService,
+    private dialogService: DialogService,
+    private router: Router,
+    public translateService: TranslateService) {
+  }
+
+  public ngOnInit(): void {
+    // TODO : show current entity name instead of id - for others than charging station c'est pas relevant
+    this.context = this.currentEntityType === Entity.TENANT ? this.centralServerService.getLoggedUser().tenantName : this.currentEntityID;
+    this.formGroup = new FormGroup({
+      id: new FormControl(),
+      entityID: new FormControl(this.currentEntityID),
+      entityType: new FormControl(this.currentEntityType),
+      // TODO: add restrictions fields
+      restrictions: new FormGroup({}),
+      name: new FormControl('',
+        Validators.compose([
+          Validators.required,
+          Validators.maxLength(100),
```

Comment:
@ClaudeROSSI Provide the same assertion in the backend Json's schema : 'pricing-definition.json'
And for all the others below.